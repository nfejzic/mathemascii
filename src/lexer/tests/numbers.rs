use super::super::*;

#[test]
fn only_number() {
    let src = "24.42";
    let mut tokens = src.tokenize();
    assert_eq!(tokens.next(), Some(Token::new("24.42", TokenKind::Number)));
}

#[test]
fn start_with_dot() {
    let src = ".42";
    let mut tokens = src.tokenize();
    assert_eq!(tokens.next(), Some(Token::new(".42", TokenKind::Number)));

    let src = ".42a";
    let mut tokens = src.tokenize();
    assert_eq!(tokens.next(), Some(Token::new(".42", TokenKind::Number)));
}

#[test]
fn start_and_stop_dot() {
    let src = ".42.3";
    let mut tokens = src.tokenize();
    assert_eq!(tokens.next(), Some(Token::new(".42", TokenKind::Number)));
}

#[test]
fn stop_at_non_digit() {
    let src = "24.42a";
    let mut tokens = src.tokenize();
    assert_eq!(tokens.next(), Some(Token::new("24.42", TokenKind::Number)));
}

#[test]
fn stop_at_dot() {
    let src = "24.42.";
    let mut tokens = src.tokenize();
    assert_eq!(tokens.next(), Some(Token::new("24.42", TokenKind::Number)));
}
