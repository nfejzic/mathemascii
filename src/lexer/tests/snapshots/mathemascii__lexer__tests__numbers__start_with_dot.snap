---
source: src/lexer/tests/numbers.rs
expression: "Snapshot((input, tokens))"
---
.42 .42a

.42
^^^ -> Number at: 0 -> 3

    .42
    ^^^ -> Number at: 4 -> 7


